File: 0645-set-mismatch/0645-set-mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        duplicate= 0
        missing= 0
        s=set()
        for i in range(len(nums)):
            val=nums[i]
            if val not in s:
                s.add(val)
            else:
                duplicate=val
        for i in range(1,len(nums)+1):
            if i not in s:
                missing=i
        return[duplicate,missing]

File: 0645-set-mismatch/0645-set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        HashSet<Integer> find = new HashSet<>();
        int repate = -1;
        for (int i = 0; i < nums.length; i++) {
            if (!find.contains(nums[i])) {
                find.add(nums[i]);
            } else {
                repate = nums[i];
            }
        }
        int missing = -1;
        for (int i = 1; i <= nums.length; i++) {
            if (!find.contains(i)) {
                missing = i;
                break;
            }
        }
        return new int []{repate,missing};
    }
}